daemons/controld/controld_remote_ra.c

```diff
@@ -219,6 +219,22 @@ purge_remote_node_attrs(int call_opt, const char *node_name)
         section = controld_section_lrm;
     }
 
+    /* Check the uptime of connection_rsc.  If it hasn't been running long
+     * enough, set purge=true regardless of what we saw above.  "Long enough"
+     * means it started running earlier than the timestamp when we noticed
+     * it went away in the first place.
+     */
+    if (connection_rsc && connection_rsc->conn) {
+        remote_ra_data_t *ra_data = connection_rsc->remote_ra_data;
+        lrmd_t *lrm = connection_rsc->conn;
+        time_t uptime = lrmd__uptime(lrm);
+        time_t now = time(NULL);
+
+        if (ra_data && uptime < now - ra_data->when_host_lost) {
```

Comment:
I'm thinking we should have a little fudge factor since it might take corosync a while to notice the host is gone. Theoretically it could depend on the token timeout but I think we can ignore that and just use something like 20 or 30s. If a remote node is up for less than that before its host goes away, I think we can reasonably ignore any attributes set in that time.

Reply:
Added 20s of fuzziness to the calculation.